File: app/src/main/java/com/travlog/android/apps/services/apirequests/NoteBody.java
package com.travlog.android.apps.services.apirequests;

import android.os.Parcel;
import android.os.Parcelable;

public class NoteBody implements Parcelable {

    public String title;
    public String memo;
    public long createdDate;
    public long modifiedDate;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.title);
        dest.writeString(this.memo);
        dest.writeLong(this.createdDate);
        dest.writeLong(this.modifiedDate);
    }

    public NoteBody() {
    }

    protected NoteBody(Parcel in) {
        this.title = in.readString();
        this.memo = in.readString();
        this.createdDate = in.readLong();
        this.modifiedDate = in.readLong();
    }

    public static final Creator<NoteBody> CREATOR = new Creator<NoteBody>() {
        @Override
        public NoteBody createFromParcel(Parcel source) {
            return new NoteBody(source);
        }

        @Override
        public NoteBody[] newArray(int size) {
            return new NoteBody[size];
        }
    };
}

File: app/src/main/java/com/travlog/android/apps/services/apirequests/NoteBody.kt
package com.travlog.android.apps.services.apirequests

import android.os.Parcel
import android.os.Parcelable
import java.util.*

class NoteBody() : Parcelable {

    var title = ""
    var memo = ""
    var createdDate = Date()
    var modifiedDate = Date()

    constructor(parcel: Parcel) : this() {
        title = parcel.readString()
        memo = parcel.readString()
        createdDate = Date(parcel.readLong())
        modifiedDate = Date(parcel.readLong())
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeString(memo)
        parcel.writeLong(createdDate.time)
        parcel.writeLong(modifiedDate.time)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<NoteBody> {

        override fun createFromParcel(parcel: Parcel): NoteBody {
            return NoteBody(parcel)
        }
        override fun newArray(size: Int): Array<NoteBody?> {
            return arrayOfNulls(size)
        }

    }

    override fun toString(): String {
        return "NoteBody(title='$title', memo='$memo', createdDate=$createdDate, modifiedDate=$modifiedDate)"
    }
}
